### PR TITLE
Load balancer module

### PR DESCRIPTION
**Pull request by brainrake**
- improve `cardano.http` and `services.http-proxy` documentation
- move default flake template to `templates/default/`
- create new flake template `temapltes/load-balancer` deploying three cardano nodes and a load balancer on vms
  - add nixos configurations for nodes
  - add load balancer configuration
  - add starter script for vms
  - detail vm network configuration
- add documentation page `getting-started/load-balancer.md`
  - explain the purpose of the module and how to use it for TLS termination and/or load balancing
  - add example for using the flake template
- add `tests/load-balancer.nix`
- improve documentation overall

### CODE REVIEW

File: templates/load-balancer/proxy.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    # # Set domain name for proxy server.
+    # # First, make sure to add DNS records for this domain and its subdomains, eg. '*.my.example.com'.
+    # # Services will be available under subdomains such as 'ogmios.my.example.com'.
+    # domainName = "my.example.com";
+
+    # # Enable HTTPS with Let's Encrypt ACME certificates.
+    # # First make sure the server is reachable via the domain name above and its subdomains.
+    # https.enable = true;
+
+    # Backend servers to proxy requests to. You may need to use IP addresses here or configure `hosts` entries or set up a DNS server, depending on your network.
```
Comment (aciceri) [inline templates/load-balancer/proxy.nix:18]: The double comment `# #` is intentional?

File: docs/getting-started/load-balancer.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+Press `Ctrl+C` to stop the machines.
+
+## Deploy to the Cloud
+
+To deploy the network of nodes and proxy to cloud providers such as AWS, Google Cloud, DigitalOcean, or Hetzner, some additional setup is required that is out of scope for this project as it depends on the deployment workflow. Here is an overview:
+
+### Deploy Cloud Infrastructure
+
+Cloud resources need to be created, for example with an Infrastructure-as-Code tool such as [OpenTofu](https://opentofu.org/) or [NixOps](https://github.com/NixOS/nixops).
```
Comment (aciceri) [inline docs/getting-started/load-balancer.md:46]: Also nixops is discouraged according to its README, I don't know if we should suggest it.

File: templates/load-balancer/proxy.nix (hunk 1 above)
Comment (brainrake) [reply templates/load-balancer/proxy.nix:18]: yes. you can uncomment the whole block and still have the comment commented

File: docs/getting-started/load-balancer.md (hunk 2 above)
Comment (brainrake) [reply docs/getting-started/load-balancer.md]: removed

**Pull request merged**